lib/natalie/compiler/pass0c.rb

```diff
@@ -38,18 +38,37 @@ def type_is_always_falsey(arg)
 
       def process_if(exp, block_with_args = nil)
         exp = exp.new(*exp.map { |e| process_atom(e) })
-        if exp.flatten.include?(:lasgn)
-          # This is required for now because
-          # ```rb
-          # x = 1 unless true # s(:if, s(:true), nil, s(:lasgn, :x, s(:lit, 1)))
-          # ```
-          # gets converted to an if-statement, that does not allocate x in the
-          # taken branch, which makes x undefined
-          return exp
-        end
         _, (boolean, *), true_case, false_case = exp
-        return true_case if type_is_always_truthy(boolean)
-        return false_case if type_is_always_falsey(boolean)
+
+        # This transforms variations of
+        # ```rb
+        # x = 1 unless true # s(:if, s(:true), nil, s(:lasgn, :x, s(:lit, 1)))
+        # ```
+        # To use `x = x`, which will be optimized away in pass1
```

Comment:
This is cool, but the comment is confusing. When I played with this just now, I get:

```
bin/natalie -e "x = 1 unless false" -d p1
s(:block, s(:var_set, :env, s(:s, :x), s(:"Value::integer", 1)))
```

...and for the opposite case:

```
bin/natalie -e "x = 1 unless true" -d p1
s(:block, s(:var_declare, :env, s(:s, :x)))
```

So is this comment really correct?

BTW I think the optimization is cool and it makes sense to have it, but I just disagree with what the comment is saying...

Reply:
I have adjusted the wording and the example to be a bit more precise.
Is it better now?